### DISCUSSION TITLE
extracting firmware information from backup dump

### DISCUSSION THREAD

**Discussion by rikrdo89** [roleoroleo/yi-hack-Allwinner-v2#693] [Q&A]
I generated a backup of my cam using an SD card as indicated in: https://github.com/roleoroleo/yi-hack-Allwinner-v2/wiki/Dump-your-backup-firmware-(SD-card) and generated the following files:
```
mtd.txt
mtdblock0.bin
mtdblock1.bin
mtdblock2.bin
mtdblock3.bin
mtdblock4.bin
mtdblock5.bin
mtdblock6.bin
mtdblock7.bin
```

Do any of these BIN files has information about the installed firmware version so I can choose the correct custom firmware to download? 

If so, is there a way to read the content of this files? I was using `dd if=mtdblock0.bin of=block0.txt  bs=1 skip=0 count=96384` but the text is barely readable.

**Comment by rikrdo89** [comment]
I actually was able to get the information by decompressing "mtdblock3.bin" using Binwalk.

```
$ binwalk -eM mtdblock3.bin
$ cat _mtdblock3.bin.extracted/squashfs-root/homever
```

I then found the same question in this other post: https://github.com/roleoroleo/yi-hack-Allwinner-v2/discussions/656#discussion-5329953

**Reply by roleoroleo** [reply]
Yes, this procecedure works properly.
You can use binwalk or unsquasfs directly.
